drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/ClasspathKieProject.java

```diff
@@ -464,14 +470,18 @@ private static String getPathForVFS(URL url) {
                 log.warn( "Found virtual file " + url + " but org.jboss.vfs.VFS is not available on the classpath" );
             }
         }
+        log.info("getPathForVFS: m=" + m + " m2=" + m2);
 
         if (m == null || m2 == null) {
             return url.getPath();
         }
 
         String path = null;
         try {
+            Object virtualFile = m2.invoke(null, url.toURI());
+            log.info("virtualFile=" + virtualFile);
             File f = (File)m.invoke( m2.invoke(null, url.toURI()) );
+            log.info("f=" + f);
```

Comment:
You can confirm that `VirtualFile.getPhysicalFile` returns file path under `deployment` directory. (e.g.  `target/cargo/configurations/wildfly23x/tmp/vfs/deployment/deploymente0f78ed559baff33/kie-pmml-7.75.0-SNAPSHOT.jar-9458e4090f0fae31/contents/META-INF/kmodule.xml`)

Actually, the `kmodule.xml` is created on the file path as a cache file. But what we want here is the `kie-pmml-7.75.0-SNAPSHOT.jar` which is under `temp` directory.